tests/tests_unit/graph/test_transformers/test_prune_graph.py

```diff
@@ -2,367 +2,87 @@
-        triples_new_predicate,
-        target_node_type,
-        namespace,
-        target_property,
-        target_property_holding_new_property,
-        expected_triples_new_predicate,
-        False,
-        id="Flatten with new predicate and keep intermediate node",
-    )
-
-    original_extended = original_triples.copy()
-
-    original_extended.extend(
-        [
-            (
-                namespace["Text-Destination-ID"],
-                target_property_holding_new_property,
-                Literal("vacuum status"),
-            ),
-            (
-                namespace["SomethingThatCanBeNode"],
-                RDF.type,
-                namespace["Node"],
-            ),
-        ]
-    )
-
-    expected_triples_new_predicate = [
-        (namespace["Device-Source-ID"], RDF.type, namespace["Device"]),
-        (
-            namespace["Device-Source-ID"],
-            namespace["vacuumStatus"],
-            namespace["SomethingThatCanBeNode"],
-        ),
-        # The intermediate target node and its properties are kept in the graph
-        (namespace["Text-Destination-ID"], RDF.type, target_node_type),
-        (
-            namespace["Text-Destination-ID"],
-            target_property,
-            Literal("SomethingThatCanBeNode"),
-        ),
-        (
-            namespace["Text-Destination-ID"],
-            target_property_holding_new_property,
-            Literal("vacuum status"),
-        ),
-        (
-            namespace["SomethingThatCanBeNode"],
-            RDF.type,
-            namespace["Node"],
-        ),
     ]
 
-    expected_triples_new_predicate = [[str(item) for item in triple] for triple in expected_triples_new_predicate]
-    expected_triples_new_predicate.sort()
-
     yield pytest.param(
-        original_extended,
-        target_node_type,
-        namespace,
-        target_property,
-        target_property_holding_new_property,
-        expected_triples_new_predicate,
-        True,
+        original_triples,
+        expected_triples,
+        removed_triples,
         id="Flatten with new predicate, literal to URIRef, delete intermediate node",
     )
 
 
-class TestAttachPropertyFromTargetToSource:
+class TestPruneGraph:
     @pytest.mark.parametrize(
-        "triples, target_node_type, namespace, "
-        "target_property, target_property_holding_new_property, expected_triples, convert_literal_to_uri",
-        list(generate_test_parameters_delete_target_node()),
+        "original_triples, expected_triples, removed_triples",
+        generate_test_parameters_unknown_types(),
     )
-    def test_two_hop_flattener_delete_connecting_node(
-        self,
-        triples: list[Triple],
-        target_node_type: URIRef,
-        namespace: Namespace,
-        target_property: URIRef,
-        target_property_holding_new_property: URIRef | None,
-        expected_triples: list[Triple],
-        convert_literal_to_uri: bool,
+    def test_prune_instances_of_unknown_type(
```

Comment:
Here is the actual new test to look at after using the new transformation base class for this transformation